vcpkg/concepts/registries.md

```diff
@@ -3,49 +3,401 @@ title: Registries concepts
+Git registries are the recommended method for implementing a custom registry.
+Since the versioning mechanism is the same used by the curated registry, Git
+registries can make use of the [`x-add-version`](../commands/add-version.md) to
+manage your version database files.
+
+See the [registries reference](../maintainers/registries.md#git-registries) for
+implementation details of Git registries.
+
+### <a name="git-add-version"> Adding a new version to a Git registry
+
+The [`x-add-version`](../commands/add-version.md) command can be used to to add
+a new port or a new version to the registry. When adding versions using this
+command there are a couple of things to keep in mind:
+
+**Remember to update the port's version**
+
+> [!IMPORTANT]
+> Always remember to update the port's version to avoid rewriting the version
+> history.
+
+When you are making changes to a port, the first step should be to increase its
+version in the `vcpkg.json` file. If your changes include an update to the
+package's [upstream](../concepts/glossary.md#upstream) version, remember to set
+the `port-version` to `0`; otherwise, remember to increase the `port-version` by
+one.
+
+**Port changes must be commited**
+
+The [`x-add-version` command](../commands/add-version.md) requires that all your
+port changes are commited to the repository before updating the version
+database.
+
+**Example: adding a new port version to a Git registry**
+
+```Console
+git add ports/foo/.
+git commit -m "Temporary commit"
+vcpkg x-add-version --x-builtin-ports-root=./ports --x-builtin-registry-versions-dir=./versions foo
+added version 1.0.0#1 to path/to/registry/versions/f-/foo.json
+added version 1.0.0#1 to path/to/registry/versions/baseline.json
+```
+
+The redirection options `--x-builtin-ports-root` and
+`--x-builtin-registry-versions-dir` should point to your registry's `ports` and
+`versions` directories respectively.
+
+Once, the `x-add-version` command runs succesfully, amend the last commit to
+include theversion database changes.
+
+```bash
+git commit --amend -m "Update foo to new version"
+```
 
 ## Filesystem registries
 
-Filesystem registries, as the name implies, live on your filesystem. They are a
-collection of ports located in a filesystem location and offer a primitive form
-of version control using a separate path per version.
+Filesystem registries are an implementation of a registry that lives on a
+filesystem. They follow the common registry structure but don't make use of Git
+to offer versioning capabilities, instead they use a primitive form of version
+control using a distinct path for each version of its ports.
+
+These type of registries are more suited to be a testing ground for your ports.
+Or to provide an alternative for registries in version control systems that are
+not Git. Since no tooling is offered to manipulate the version database files,
+this method is not recommended for large collections of ports.
```

Comment:
```suggestion
Filesystem registries are an implementation of a registry that lives on a
filesystem. They follow the common registry structure but don't make use of Git
to offer versioning capabilities. Instead, they use a primitive form of version
control using a distinct path for each version of its ports.

These types of registries are more suited to be a testing ground for your ports or to provide an alternative for registries in version control systems that are
not Git. Since no tooling is offered to manipulate the version database files,
this method is not recommended for large collections of ports.
```